### PR TITLE
Add Forms

### PR DESCRIPTION
**Pull request by odhiambo-ed**
# Blog app - Add forms

## Description

In this milestone, I added forms to the Blog app.

## Project requirements

-  Create a method called `current_user` in your ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
    -  Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,2 +1,16 @@
 class CommentsController < ApplicationController
+  def create
+    @publisher = User.find(params[:user_id])
```
Comment (KDlamini) [inline app/controllers/comments_controller.rb:7]: Please make sure that the comments controller has the `new`  action for creating a new comment.
 i.e
``` 
def new
...
  end
```

File: config/routes.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-  # get 'about', to: 'pages#about'
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  get '/', to: 'users#index'
 
-  resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+  resources :users, only: %i[index show] do
+    resources :posts, only: %i[index show new create] do
+      resources :comments, only: [:create]
+    end
```
Comment (KDlamini) [inline config/routes.rb]: - Kindly include the `new` route for comments as well.

File: config/routes.rb [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
   end
-  # Defines the root path route ("/")
-  # root "articles#index"
-end
+
+  put 'user/:user_id/post/:id/like', to: 'posts#like', as: 'like'
```
Comment (KDlamini) [inline config/routes.rb]: The method for creating a new like should be `POST` not `PUT`. Kindly use `POST` for the likes route.

Comment (KDlamini) [inline config/routes.rb:11]: The method for creating a new like should be `POST` not `PUT`. Kindly use `POST` for the likes route.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @white3d  🙋‍♀️,  

Great work so far! 💯 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 
✔️  Correct PR description
✔️  Commit messages are meaningful
✔️  No linter errors

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @white3d,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**